core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -104,6 +127,29 @@ ManifestGroup caseSensitive(boolean newCaseSensitive) {
   }
 
   /**
+   * Returns a iterable of scan tasks. It is safe to add entries of this iterable
+   * to a collection as {@link DataFile} in each {@link FileScanTask} is defensively
+   * copied.
+   * @return a {@link CloseableIterable} of {@link FileScanTask}
+   */
+  public CloseableIterable<FileScanTask> scanTasks() {
+    Iterable<CloseableIterable<FileScanTask>> tasks = entries((manifest, entries) -> {
+      PartitionSpec spec = specsById.get(manifest.partitionSpecId());
+      String schemaString = SchemaParser.toJson(spec.schema());
+      String specString = PartitionSpecParser.toJson(spec);
+      ResidualEvaluator residuals = ResidualEvaluator.of(spec, dataFilter, caseSensitive);
```

Comment:
Minor for follow-up: it would be nice to cache residual evaluators by spec ID so that we don't create one per manifest.